libs/content/workspace/src/tab/svg_editor/selection/mod.rs

```diff
@@ -174,63 +196,64 @@ impl Selection {
         }
 
         if self.laso_rect.is_none() {
-            self.selection_rect =
-                SelectionRectContainer::new(&self.selected_elements, working_rect, buffer);
+            self.selection_rect = SelectionRectContainer::new(&self.selected_elements, buffer);
         }
 
         let mut intent = None;
         if let Some(r) = &self.selection_rect {
-            r.show(ui);
-            intent = r.get_cursor_icon(pos);
+            r.show(ui, painter);
+            if let Some(p) = pos {
+                if painter.clip_rect().contains(p) {
+                    intent = r.get_cursor_icon(p);
+                }
+            }
         }
 
         if ui.input(|r| r.pointer.primary_released()) {
-            end_translation(buffer, history, &mut self.selected_elements, pos, true);
+            if let Some(p) = pos {
+                end_translation(buffer, history, &mut self.selected_elements, p, true);
+            }
             self.current_op = SelectionOperation::Idle;
-        } else if ui.input(|r| r.pointer.primary_clicked()) {
-            end_translation(buffer, history, &mut self.selected_elements, pos, false);
+        } else if ui.input(|r| r.pointer.primary_clicked()) && pos.is_some() {
+            end_translation(buffer, history, &mut self.selected_elements, pos.unwrap(), false);
         } else if ui.input(|r| r.pointer.primary_down()) {
             if matches!(self.current_op, SelectionOperation::Idle) {
                 if let Some(r) = &mut intent {
                     self.current_op = r.current_op;
                     ui.output_mut(|w| w.cursor_icon = r.cursor_icon);
                 }
             }
-
-            match self.current_op {
-                SelectionOperation::Translation => {
-                    self.selected_elements.iter_mut().for_each(|selection| {
-                        if let Some(el) = buffer.elements.get_mut(&selection.id) {
-                            let transform = Transform::identity().post_translate(
-                                pos.x - selection.prev_pos.x,
-                                pos.y - selection.prev_pos.y,
-                            );
-                            selection.transform = selection.transform.post_concat(transform);
-                            match el {
-                                parser::Element::Path(p) => {
-                                    p.data.apply_transform(u_transform_to_bezier(&transform));
-                                }
-                                parser::Element::Image(img) => img.apply_transform(transform),
-                                parser::Element::Text(_) => todo!(),
+            if pos.is_some() {
```

Comment:
nit:
```suggestion
            if let Some(p) = pos {
```

Reply:
non-nit imo this would eliminate an unwrap below